pkg/auth/manager/oidcmapping/oidcmapping.go

```diff
@@ -199,43 +198,72 @@ func (am *mgr) Authenticate(ctx context.Context, clientID, clientSecret string)
 		gid, _ = claims[am.c.GIDClaim].(float64)
 	}
 
-	gwc, err := pool.GetUserProviderServiceClient(am.c.UserProviderSvc)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "oidcmapping: error getting gateway grpc client")
-	}
-
 	userID := &user.UserId{
 		OpaqueId: "",
 		Idp:      "",
-		Type:     user.UserType_USER_TYPE_PRIMARY,
+		Type:     user.UserType_USER_TYPE_INVALID,
 	}
 
-	if username != "" {
-		getUserByClaimResp, err := gwc.GetUserByClaim(ctx, &user.GetUserByClaimRequest{
+	var username string
+	if len(am.oidcUsersMapping) > 0 {
```

Comment:
I wouldn't recommend refactoring both the oidc drivers into one. This one has a really specific use case and I would really want to keep it separate from the other generic one.

Reply:
As discussed I'll try to factorize the mapping-specific code so as to leave the `Authorize` function the cleanest and most generic possible.